snippets/binarySearch.md

```diff
@@ -0,0 +1,22 @@
+### binarySearch
+
+Use recursion to perform a logarithmic search similar to `.indexOf()` that finds the index of a value within an array. The two differences being 
+1. Operation only works with sorted arrays
+2. Offers a major performance boost when compared to a linear search or `.indexOf()` 
+
+```js
+const binarySearch = (arr, val, start = 0, end = arr.length - 1) => {
+  if (start > end) return -1;
+  const mid = Math.floor((start + end) / 2);
+  const target = arr[mid];
+  if (target > val) return binarySearch(arr, val, start, mid-1);
+  if (target < val) return binarySearch(arr, val, mid+1, end);
+  return mid;
```

Comment:
Lines 12-14 could be one long ternary operator statement. Also, is there a need to assign `target` in the first place? You are not using it anywhere except these conditionals. I would argue the same about `mid`, but that would make the code slightly less readable and you would have to calculate it like 5 times, which is a performance hit on a performance-centric snippet, so I think that should remain as-is.